server/src/main/java/io/seata/server/storage/file/lock/FileLocker.java

```diff
@@ -101,7 +101,6 @@ public boolean acquireLock(List<RowLock> rowLocks, boolean autoCommit, boolean s
                 LOGGER.info("Global lock on [" + tableName + ":" + pk + "] is holding by " + previousLockBranchSession.getBranchId());
                 try {
                     if (CollectionUtils.isNotEmpty(bucketHolder)) {
```

Comment:
use whether the branchSession is in AT or not, instead of judging whether the bucketHolder is empty.

Reply:
> use whether the branchSession is in AT or not, instead of judging whether the bucketHolder is empty.

假设这个分支再竞争第一个锁就失败了,那么就无需进行unlock不是吗?
Suppose this branch fails to compete for the first lock, then there is no need to unlock it, right?

Reply:
明白了，按这么说其实filelocker的存在其实是确定锁被哪个branch拿走了，只做分配锁动作，而归还锁（释放）交由branch来，就有面向对象的感觉，锁，分支，锁被分支拿走了，要分支归还，而现在改的有点面向过程了。

Reply:
done